File: examples/javascript/list-objects.mjs
// Hippius S3 — List objects with AWS SDK v3
// Docs: /storage/s3/javascript

import "dotenv/config";
import { S3Client, ListObjectsV2Command } from "@aws-sdk/client-s3";

const s3 = new S3Client({
  endpoint: "https://s3.hippius.com",
  region: "decentralized",
  credentials: {
    accessKeyId: process.env.HIPPIUS_ACCESS_KEY,
    secretAccessKey: process.env.HIPPIUS_SECRET_KEY,
  },
  forcePathStyle: true,
});

const bucketName = "my-bucket";

const response = await s3.send(new ListObjectsV2Command({
  Bucket: bucketName,
}));

for (const obj of response.Contents ?? []) {
  console.log(`${obj.Key}  (${obj.Size} bytes)`);
}

File: examples/python/list_objects.py
# Hippius S3 — List objects with boto3
# Docs: /storage/s3/python

import os
from dotenv import load_dotenv
import boto3
from botocore.config import Config

load_dotenv()

s3 = boto3.client(
    "s3",
    endpoint_url="https://s3.hippius.com",
    aws_access_key_id=os.environ["HIPPIUS_ACCESS_KEY"],
    aws_secret_access_key=os.environ["HIPPIUS_SECRET_KEY"],
    region_name="decentralized",
    config=Config(
        signature_version="s3v4",
        s3={"addressing_style": "path"},
    ),
)

bucket_name = "my-bucket"

response = s3.list_objects_v2(Bucket=bucket_name)
for obj in response.get("Contents", []):
    print(f"{obj['Key']}  ({obj['Size']} bytes)")
